src/util/string.h

```diff
@@ -87,4 +87,9 @@ template <typename T1, size_t PREFIX_LEN>
            std::equal(std::begin(prefix), std::end(prefix), std::begin(obj));
 }
 
+/**
+ * User-defined literal for byte constants.
+ */
+constexpr uint8_t operator ""_u8(unsigned long long byte) { return byte; }
```

Comment:
Maybe we should add some overflow assert here? it would be nice if `256_u8` won't compile.
the downside is that this function might be used outside of constexpr, but that's very unlikely

Reply:
ie `assert(byte <= std::numeric_limits<std::uint8_t>::max());`

Reply:
> Maybe we should add some overflow assert here? it would be nice if `256_u8` won't compile.

Agree...

> ie `assert(byte <= std::numeric_limits<uint8_t>::max());`

... but this won't work, unfortunately.

Reply:
Why not? 
https://godbolt.org/z/hTjch7

Reply:
It is not a compile-time error.

Reply:
You're right, idk how I didn't realized that

Reply:
You can use runtime asserts in constexpr functions. It just means invocations that would trigger that revert to being runtime evaluated. If the result is assigned to a constexpr variable, it'll error at compile time.

Reply:
Why does c++ not allow constexpr params, so that a static_assert could be used here?

Reply:
:man_shrugging:

Reply:
You... could.

```c++
constexpr uint8_t operator ""_u8(unsigned long long byte)
{
    constexpr unsigned long long v = byte;
    static_assert(v < 256);
    return v;
}
```

Reply:
C++20 adds (`consteval`)[https://en.cppreference.com/w/cpp/language/consteval] which would be useful here.

Reply:
I'd probably prefer `std::make_array` to be backported. https://en.cppreference.com/w/cpp/experimental/make_array

This would allow to write `MakeArray<uint8_t>(1,2,3)`

Reply:
[`std::to_array`](https://en.cppreference.com/w/cpp/container/array/to_array) from C++20 ?